panic on startup: "could not save account data"

### Background information

- **Dendrite version or git SHA**: 0.3.8
- **Monolith or Polylith?**: monolith
- **SQLite3 or Postgres?**: sqlite3
- **Running in Docker?**: nope
- **`go version`**: 1.15.7 linux/amd64

### Description

after starting my dendrite service, it panics with the following log:
```
time=(time of log) level=panic msg="could not save account data" func="onMessage\n\t" file=" [github.com/matrix-org/dendrite/syncapi/consumers/clientapi.go:97]" error="UNIQUE constraint failed: syncapi_account_data_type.id" room_id=(matrix room id) type=m.fully_read
panic: (*logrus.Entry) 0xc000189b90
goroutine 93 [running]:
github.com/sirupsen/logrus.Entry.log(0xc0000dac40, 0xc0003c78c0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        github.com/sirupsen/logrus@v1.7.0/entry.go:255 +0x325
github.com/sirupsen/logrus.(*Entry).Log(0xc000189b20, 0x0, 0xc000377b08, 0x1, 0x1)
        github.com/sirupsen/logrus@v1.7.0/entry.go:283 +0xf0
github.com/sirupsen/logrus.(*Entry).Logf(0xc000189b20, 0xc000000000, 0x561249435606, 0x1b, 0x0, 0x0, 0x0)
        github.com/sirupsen/logrus@v1.7.0/entry.go:329 +0xe5
github.com/sirupsen/logrus.(*Entry).Panicf(...)
        github.com/sirupsen/logrus@v1.7.0/entry.go:367
github.com/matrix-org/dendrite/syncapi/consumers.(*OutputClientDataConsumer).onMessage(0xc0002c47e0, 0xc000321400, 0x1, 0x101)
        github.com/matrix-org/dendrite/syncapi/consumers/clientapi.go:97 +0x725
github.com/matrix-org/dendrite/internal.(*ContinualConsumer).consumePartition(0xc0001946c0, 0x56124983e360, 0xc00017af20)
        github.com/matrix-org/dendrite/internal/consumers.go:126 +0xe2
created by github.com/matrix-org/dendrite/internal.(*ContinualConsumer).StartOffsets
        github.com/matrix-org/dendrite/internal/consumers.go:107 +0x545
```
unfortunately i don't have much in terms of steps for reproducing this from a clean db, but i hope this is enough to help.

--- a/syncapi/storage/postgres/account_data_table.go
+++ b/syncapi/storage/postgres/account_data_table.go
@@ -53,7 +53,7 @@ CREATE UNIQUE INDEX IF NOT EXISTS syncapi_account_data_id_idx ON syncapi_account
 const insertAccountDataSQL = "" +
 	"INSERT INTO syncapi_account_data_type (user_id, room_id, type) VALUES ($1, $2, $3)" +
 	" ON CONFLICT ON CONSTRAINT syncapi_account_data_unique" +
-	" DO UPDATE SET id = EXCLUDED.id" +
+	" DO UPDATE SET id = nextval('syncapi_stream_id')" +
 	" RETURNING id"
 
 const selectAccountDataInRangeSQL = "" +

--- a/syncapi/storage/sqlite3/account_data_table.go
+++ b/syncapi/storage/sqlite3/account_data_table.go
@@ -39,7 +39,7 @@ CREATE TABLE IF NOT EXISTS syncapi_account_data_type (
 const insertAccountDataSQL = "" +
 	"INSERT INTO syncapi_account_data_type (id, user_id, room_id, type) VALUES ($1, $2, $3, $4)" +
 	" ON CONFLICT (user_id, room_id, type) DO UPDATE" +
-	" SET id = EXCLUDED.id"
+	" SET id = $5"
 
 const selectAccountDataInRangeSQL = "" +
 	"SELECT room_id, type FROM syncapi_account_data_type" +
@@ -86,7 +86,7 @@ func (s *accountDataStatements) InsertAccountData(
 	if err != nil {
 		return
 	}
-	_, err = sqlutil.TxStmt(txn, s.insertAccountDataStmt).ExecContext(ctx, pos, userID, roomID, dataType)
+	_, err = sqlutil.TxStmt(txn, s.insertAccountDataStmt).ExecContext(ctx, pos, userID, roomID, dataType, pos)
 	return
 }
 

